service/frontend/fx.go

```diff
@@ -367,11 +377,20 @@ func RateLimitInterceptorProvider(
 	)
 }
 
+func getNamespaceRateFnWithMetrics(rateFn func(ns string) float64, handler metrics.Handler) quotas.NamespaceRateFn {
+	return func(ns string) float64 {
+		rate := rateFn(ns)
+		metrics.NamespaceHostRPSLimit.With(handler.WithTags(metrics.NamespaceTag(ns))).Record(rate)
```

Comment:
Unfortunately, this metric will generate high metric cardinality (|pods| x |namespaces|), and we have several clusters that cannot really afford that, so I think it should be omitted for now. 

Besides, the motivation to get these limits into metrics is to help determine if the cluster needs to be scaled for the health of the cluster. It's not as important to get forewarning on a namespace approaching its limits because it doesn't impact the cluster health.

Reply:
Got it. Removed namespace limit metrics.